pkgs/build-support/trivial-builders/default.nix

```diff
@@ -625,18 +625,28 @@ rec {
 
   # Docs in doc/build-helpers/trivial-build-helpers.chapter.md
   # See https://nixos.org/manual/nixpkgs/unstable/#trivial-builder-writeReferencesToFile
-  writeReferencesToFile = path: runCommand "runtime-deps"
+  writeReferencesToFile = path: writeReferenceClosureToFile [ path ];
+
+  # Docs in doc/build-helpers/trivial-build-helpers.chapter.md
+  # See https://nixos.org/manual/nixpkgs/unstable/#trivial-builder-writeReferenceClosureToFile
+  writeReferenceClosureToFile = paths: runCommand "runtime-deps"
     {
-      exportReferencesGraph = [ "graph" path ];
+      __structuredAttrs = true; # Get the cleaner exportReferencesGraph interface
+      exportReferencesGraph.graph = paths;
+      nativeBuildInputs = [ nix ];
     }
     ''
-      touch $out
-      while read path; do
-        echo $path >> $out
-        read dummy
-        read nrRefs
-        for ((i = 0; i < nrRefs; i++)); do read ref; done
-      done < graph
+      nix --extra-experimental-features "nix-command" \
+        --offline \
+        eval --eval-store "$PWD" --store "$PWD" --impure --expr ${lib.escapeShellArg ''
+          let
+            lib = import ${nixpkgs}/lib;
+            rawAttrs = lib.importJSON ./.attrs.json;
+            references = lib.sort lib.lessThan (map (lib.getAttr "path") rawAttrs.graph);
+          in
+          lib.concatStringsSep "\n" references
+        ''} \
+        --raw > "$out"
```

Comment:
I think it's better not to introduce a dependency on nix... As mentioned above, `closure-info` contains examples of using `jq` and `referencesByPopularity` - of using python's `json` - to read the output of `exportReferencesGraph`

Reply:
Alright.

Reply:
Addressed by parsing and sorting with jq.